packages/core/src/controllers/OnRampController.ts

```diff
@@ -93,6 +100,15 @@ export const OnRampController = {
   },
 
   setSelectedProvider(provider: OnRampProvider | null) {
+    if (provider && provider.name === 'meld') {
+      const pubKey =
+        process.env['NODE_ENV'] === 'production' ? MELD_PROD_PUBLIC_KEY : MELD_DEV_PUBLIC_KEY
+      provider.url += pubKey
+      const currency = ChainController.state.activeChain === 'solana' ? 'SOL' : 'USDC'
+      provider.url += `&destinationCurrencyCode=${currency}`
+      const address = AccountController.state.address
+      provider.url += `&walletAddress=${address}`
+    }
```

Comment:
Consider using the `URL` class to construct the URL. It offers a more maintainable and readable approach for handling query parameters.

```suggestion
if (provider && provider.name === 'meld') {
  const pubKey = process.env['NODE_ENV'] === 'production' ? MELD_PROD_PUBLIC_KEY : MELD_DEV_PUBLIC_KEY;
  const currency = ChainController.state.activeChain === 'solana' ? 'SOL' : 'USDC';
  const address = AccountController.state.address;

  const url = new URL(provider.url);
  url.searchParams.append('publicKey', pubKey);
  url.searchParams.append('destinationCurrencyCode', currency);
  url.searchParams.append('walletAddress', address);

  provider.url = url.toString();
}
```

Reply:
+ use `ConstantsUtil.Chain.SOLANA` for `"solana"` hard coded value